### ISSUE TITLE
Training with LoRA: Train loss does not converge

### ISSUE THREAD

**Issue by priprapre** [huggingface/diffusers#2482]
I am running the [`./examples/text_to_image/train_text_to_image.py`](https://github.com/huggingface/diffusers/tree/main/examples/text_to_image#training-with-lora) script on my custom dataset (also tried with the pokemon dataset), it definitely learns something, it gets the domain of the dataset and generate similar images, but the train loss doesn't actually decrease. I feel like it learns very quickly in the first few steps and then there is no real improvement.

I have also checked [the LoRA article](https://huggingface.co/blog/lora) of HuggingFace and [the W&B dashboard](https://wandb.ai/pcuenq/text2image-fine-tune/runs/b4k1w0tn?workspace=user-pcuenq) of Pokemon training mentioned there, it also does not seem to converge. The images seem to evolve but I am also not sure if they actually converge to some point, it seems a bit like a random walk in the image space.

Does anyone have an idea what might be the issue and how to solve it? Thanks in advance!

**Comment by haofanwang** [comment]
It's normal that the model converge quickly in about 100-1000 steps. More steps may lead to serious overfitting. I think the codebase should be fine, as I have some interesting projects based on it. One thing you can try is to increase the rank of LoRA to 128 or 256, the default is only 4, which makes the overfitting more quickly. @priprapre

**Comment by patrickvonplaten** [comment]
cc @sayakpaul maybe

**Comment by sayakpaul** [comment]
The loss surface is relatively uninformative (much like the initial GAN days, which was later resolved by things like [WGAN with GP](https://paperswithcode.com/paper/improved-training-of-wasserstein-gans)). One of the reasons why it might be happening is because of how timesteps are sampled during the process. In one training step, it could be 2, in the following, training step it could be 200. Thanks to @patil-suraj for this probable intuition. 

I'd suggest relying on some qualitative validation during the process of fine-tuning as we do in our example training scripts. An example is here:

https://github.com/huggingface/diffusers/blob/bbab8553224d12f7fd58b0e65b0daf899769ef0b/examples/dreambooth/train_dreambooth_lora.py#L950-L989
